entra_id/token_manager.go

```diff
@@ -0,0 +1,129 @@
+package entra_id
```

Comment:
In the official [go styleguide](https://google.github.io/styleguide/go/decisions.html#package-names) it is suggested to not use `_` in package names. Let's rename the package to `entraid` or even if we end up with a different name once we move the repo, we can follow the styleguide.